src/GraphQLParser/LexerContext.cs

```diff
@@ -181,16 +179,24 @@ private Token ReadComment()
             );
         }
 
+        // TODO: this method can still be optimized no not allocate at all if block string:
+        //
+        // 1) not too long
+        // 2) has no escape sequences
+        // 3) has no '\r` characters
```

Comment:
```suggestion
        // 1) not too long
        // 2) has no escape sequences
        // 3) has no '\r' characters
        // 4) has no initial whitespace on each line, ignoring the first line (or, has no '\n' characters)
```

Reply:
Keep in mind that:

`\"""  test\"""` equals `"  test"` but `\"""\n  test\"""` equals `"\ntest"` (whitespace removed from second sample)